cogify/handler.py

```diff
@@ -11,68 +11,103 @@
 import boto3
 from typing import Optional
 import configparser
+import argparse
+import sys
+
 config = configparser.ConfigParser()
-config.read('example.ini')
-s3 = boto3.client('s3')
+config.read("example.ini")
+s3 = boto3.client(
+    "s3",
+    aws_access_key_id=os.environ.get('AWS_ACCESS_KEY_ID'),
+    aws_secret_access_key=os.environ.get('AWS_SECRET_ACCESS_KEY'),
+)
 
 # Set COG inputs
 output_profile = cog_profiles.get(
     "deflate"
 )  # if the files aren't uint8, this will need to be changed
 output_profile["blockxsize"] = 256
 output_profile["blockysize"] = 256
-output_bucket = config['DEFAULT']['output_bucket']
-output_dir = config['DEFAULT']['output_dir']
+output_bucket = config["DEFAULT"]["output_bucket"]
+output_dir = config["DEFAULT"]["output_dir"]
+
+parser = argparse.ArgumentParser(description="Cogify handler workflow")
```

Comment:
I was steering away from using args since converting this to work as a lambda necessitated that arguments can be passed into the handler as a dictionary.

I think you're right that those 2 approaches (passing command line arguments for local development and passing a dictionary to the handler object for the AWS lambda) are not mutually exclusive ways of calling the function code. My only concern would be ensuring that this doesn't result in different results between testing locally and deploying as a lambda function. I could see how someone testing changes locally modifies the command line arguments and sees that the handler code works locally for those command line arguments but then throws an error when deployed as a lambda function because they didn't update the handler implementation. What do you think about this concern?